src/HttpReply.cc

```diff
@@ -259,6 +260,16 @@ HttpReply::validatorsMatch(HttpReply const * otherRep) const
     return 1;
 }
 
+static bool
+CheckReplyHeaderSize(const HttpReplyPointer &reply, size_t &size)
+{
+    MemBuf mb;
+    mb.init();
+    reply->header.packInto(&mb);
```

Comment:
The packing and size calculation is performed in other places, e.g., before sending the response (Http::Stream::sendStartOfMessage()). Probably we can optimize this operation, caching somewhere this result. I have not investigated whether it is worth/easy to do.